spec/technical_analysis/indicators/wma_spec.rb

```diff
@@ -0,0 +1,115 @@
+require 'technical-analysis'
+require 'spec_helper'
+
+describe 'Indicators' do
+  describe "WMA" do
+    input_data = SpecHelper.get_test_data(:close, date_time_key: :timestep)
+    indicator = TechnicalAnalysis::Wma
+
+    describe 'Weighted Moving Average' do
+      it 'Calculates WMA (5 day)' do
+        output = indicator.calculate(input_data, period: 5, price_key: :close, date_time_key: :timestep)
+        normalized_output = output.map(&:to_hash)
+
+        expected_output = [
+          {:date_time=>"2019-01-09T00:00:00.000Z", :wma=>150.13666666666666},
+          {:date_time=>"2019-01-08T00:00:00.000Z", :wma=>148.83666666666667},
+          {:date_time=>"2019-01-07T00:00:00.000Z", :wma=>148.856},
+          {:date_time=>"2019-01-04T00:00:00.000Z", :wma=>150.36866666666666},
+          {:date_time=>"2019-01-03T00:00:00.000Z", :wma=>152.29733333333334},
+          {:date_time=>"2019-01-02T00:00:00.000Z", :wma=>157.248},
+          {:date_time=>"2018-12-31T00:00:00.000Z", :wma=>156.216},
+          {:date_time=>"2018-12-28T00:00:00.000Z", :wma=>154.77666666666667},
+          {:date_time=>"2018-12-27T00:00:00.000Z", :wma=>153.88066666666666},
+          {:date_time=>"2018-12-26T00:00:00.000Z", :wma=>153.32733333333334},
+          {:date_time=>"2018-12-24T00:00:00.000Z", :wma=>153.02733333333333},
+          {:date_time=>"2018-12-21T00:00:00.000Z", :wma=>157.31466666666665},
+          {:date_time=>"2018-12-20T00:00:00.000Z", :wma=>161.28533333333334},
+          {:date_time=>"2018-12-19T00:00:00.000Z", :wma=>164.164},
+          {:date_time=>"2018-12-18T00:00:00.000Z", :wma=>166.23666666666665},
```

Comment:
I would suggest randomizing the order of some of these (just swap several lines) so you can ensure that your incoming data is sorted properly.

Reply:
I believe the test would fail if I did that.

Reply:
I just checked and indeed it failed when I swapped the order of the results.  The order is important and the test should fail if the order is different.